container/concurrent/lrucache.go

```diff
@@ -0,0 +1,91 @@
+package concurrent
+
+import (
+	"sync"
+	"github.com/dropbox/godropbox/container/lrucache"
+)
+
+type LRUCache interface {
+	GetMultiple(keys []string) []CacheResult
+	Get(key string) (v interface{}, found bool)
+	Set(key string, v interface{})
+	SetMultiple(keyValues map[string]interface{})
+	Delete(keys ...string)
+	Clear()
+}
+
+type CacheResult struct {
+	V     interface{}
+	Found bool
+}
+
+type concurrentLruCacheImp struct {
+	cache *lrucache.LRUCache
+	lock  sync.RWMutex
+	size  int
+}
+
+func NewLRUCache(size int) LRUCache {
+	cache := lrucache.New(size)
+
+	return &concurrentLruCacheImp{
+		cache: cache,
+		size:  size,
+	}
+}
+
+func (p *concurrentLruCacheImp) Get(key string) (v interface{}, found bool) {
+	p.lock.RLock()
```

Comment:
Good catch....I've added unit-tests too now. I guess RLock is re-entrant so, it's not as bad as a Lock